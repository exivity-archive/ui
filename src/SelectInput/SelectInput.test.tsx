import * as React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import { SelectInput } from '.'

test('SelectInput snapshot', () => {
  const selectInput = renderer.create(
    <SelectInput value='TextInputWIthIcon' onChange={jest.fn()} />
  )

  expect(selectInput).toMatchSnapshot()
})

test('value gets changed after an on change event', () => {
  const initialValue = 'initialValue'
  const updatedValue = 'changed'

  const onChangeMock = jest.fn(x => x)

  const selectInput = mount(<SelectInput value={initialValue} onChange={onChangeMock} />)

  selectInput
    .find('input')
    .simulate('change', { target: { value: updatedValue } })

  expect(onChangeMock.mock.results[0].value).toBe(updatedValue)
})
