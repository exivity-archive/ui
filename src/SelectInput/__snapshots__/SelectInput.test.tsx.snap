// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput snapshot 1`] = `
.c3 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  font-size: 16px;
  display: block;
  box-sizing: border-box;
  width: 100%;
  padding: calc(0.5em - 1px) 0.5em;
  border-radius: 4px;
  outline: 0;
  border: 0;
  border: 1px solid #eee;
  background-color: #eee;
  --focus-color: 0,168,216;
}

.c3:hover {
  border-bottom: 1px solid rgba(var(--focus-color),0.5);
}

.c3:focus {
  border-bottom: 1px solid rgba(var(--focus-color),1);
}

.c3::-webkit-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c3::-moz-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c3:-ms-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c3::placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c3[disabled] {
  cursor: not-allowed;
  box-shadow: inset 0 0 999em rgba(128,128,128,0.2);
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: column;
  -webkit-justify-content: column;
  -ms-flex-pack: column;
  justify-content: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  padding-right: 2.2em;
}

.c4 {
  font-size: 16px;
  height: 1em;
  line-height: 1em;
  text-transform: none;
  position: absolute;
  font-size: 20px;
  cursor: not-allowed;
  right: 0.5em;
}

.c1[disabled] {
  cursor: pointer;
  box-shadow: none;
}

.c1 span {
  cursor: pointer;
}

<div
  className="c0"
>
  <input
    className="c1 c2 c3"
    disabled={true}
    onChange={[Function]}
    type="text"
    value="TextInputWIthIcon"
  />
  <span
    className="c1 c4"
    disabled={true}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
      />
    </svg>
  </span>
</div>
`;
